packages/instantsearch.js/src/lib/utils/hydrateSearchClient.ts

```diff
@@ -56,9 +58,11 @@ export function hydrateSearchClient(
           Object.keys(results).reduce(
             (acc, key) =>
               acc.concat(
-                results[key].results.map((request) => ({
-                  indexName: request.index,
-                  params: request.params,
+                results[key].results.map((result) => ({
+                  indexName: result.index,
+                  // We normalize the params received from the server as they can
+                  // be serialized differently depending on the engine.
+                  params: serializeQueryParameters(qs.parse(result.params)),
```

Comment:
while this helps in some cases (different normalisation), this isn't actually what the client does:

https://github.com/algolia/algoliasearch-client-javascript/blob/075b9fa2eeb83b12700d7c9c402f0520ab04681c/packages/transporter/src/createTransporter.ts#L88-L95

it uses the params from the request to create the key, and therefore no normalisation is needed. Obviously it's annoying that we need to re-implement createMappedRequestOptions instead, but I think it's a more correct solution.

Reply:
The problem is we don't have access to the request, but yeah the shape shouldn't be different